drivers/adc/adc_mcux_adc16.c

```diff
@@ -278,34 +413,85 @@ static const struct adc_driver_api mcux_adc16_driver_api = {
 #endif
 };
 
-#define ACD16_MCUX_INIT(n)						\
-	static void mcux_adc16_config_func_##n(const struct device *dev); \
-									\
+#ifdef CONFIG_ADC_MCUX_ADC16_ENABLE_EDMA
+#define ACD16_MCUX_INIT(n)											\
 	static const struct mcux_adc16_config mcux_adc16_config_##n = {	\
-		.base = (ADC_Type *)DT_INST_REG_ADDR(n),		\
-		.irq_config_func = mcux_adc16_config_func_##n,		\
-		.channel_mux_b = DT_INST_PROP(n, channel_mux_b),	\
-	};								\
-									\
-	static struct mcux_adc16_data mcux_adc16_data_##n = {		\
-		ADC_CONTEXT_INIT_TIMER(mcux_adc16_data_##n, ctx),	\
-		ADC_CONTEXT_INIT_LOCK(mcux_adc16_data_##n, ctx),	\
-		ADC_CONTEXT_INIT_SYNC(mcux_adc16_data_##n, ctx),	\
-	};								\
-									\
-	DEVICE_DT_INST_DEFINE(n, &mcux_adc16_init,			\
-			    device_pm_control_nop, &mcux_adc16_data_##n,\
-			    &mcux_adc16_config_##n, POST_KERNEL,	\
-			    CONFIG_KERNEL_INIT_PRIORITY_DEVICE,		\
-			    &mcux_adc16_driver_api);			\
-									\
-	static void mcux_adc16_config_func_##n(const struct device *dev) \
-	{								\
-		IRQ_CONNECT(DT_INST_IRQN(n), DT_INST_IRQ(n, priority),	\
-			    mcux_adc16_isr,				\
-			    DEVICE_DT_INST_GET(n), 0);			\
-									\
-		irq_enable(DT_INST_IRQN(n));				\
+		.base = (ADC_Type *)DT_INST_REG_ADDR(n),					\
+		.channel_mux_b = DT_INST_PROP(n, channel_mux_b),			\
+		.clk_source = UTIL_AND(DT_INST_NODE_HAS_PROP(n, nxp_clk_source),	\
+				       DT_INST_PROP(n, nxp_clk_source)),			\
+		.nxp_long_sample =											\
+			UTIL_AND(DT_INST_NODE_HAS_PROP(n, nxp_long_sample),		\
+				 DT_INST_PROP(n, nxp_long_sample)),					\
+		.nxp_high_speed =											\
+			UTIL_AND(DT_INST_NODE_HAS_PROP(n, nxp_high_speed),		\
+				 DT_INST_PROP(n, nxp_high_speed)),					\
+		.nxp_continuous_convert = UTIL_AND(							\
+			DT_INST_NODE_HAS_PROP(n, nxp_continuous_convert),		\
+			DT_INST_PROP(n, nxp_continuous_convert)),				\
```

Comment:
`DT_INST_PROP(n, nxp_continuous_convert)`

Reply:
ok

Reply:
And this one

Reply:
ok